Sources/Workspace/InitPackage.swift

```diff
@@ -62,102 +48,161 @@ public final class InitPackage {
             self.packageType = packageType
         }
     }
+    
+    public enum MakePackageMode {
+        case `initialize`
+        case create
+    }
 
     /// A block that will be called to report progress during package creation
     public var progressReporter: ((String) -> Void)?
 
     /// Where to create the new package
     let destinationPath: AbsolutePath
-
-    /// Package Template
-    var packageTemplate: PackageTemplate
     
-    /// The options for package to create.
-    let options: InitPackageOptions
-
-    /// The name of the package to create.
-    let pkgname: String
-
-    /// The name of the target to create.
-    var moduleName: String
-
-    /// The name of the type to create (within the package).
-    var typeName: String {
-        return moduleName
+    let fileSystem: FileSystem
+    
+    /// Configuration path for templates
+    let configPath: AbsolutePath
+    
+    let packageName: String
+    
+    let mode: MakePackageMode
+    
+    let packageType: InitPackage.PackageType
+    
+    let packageTemplateName: String?
+    
+    private var moduleName: String {
+        get {
+            packageName.spm_mangledToC99ExtendedIdentifier()
+        }
     }
-
-    /// Create an instance that can create a package with given arguments.
-    public convenience init(
-        name: String,
-        destinationPath: AbsolutePath,
-        packageTemplate: PackageTemplate
-    ) throws {
-        try self.init(
-            name: name,
-            destinationPath: destinationPath,
-            options: InitPackageOptions(),
-            packageTemplate: packageTemplate
-        )
+    
+    private var typeName: String {
+        get {
+            moduleName
+        }
     }
     
-    public convenience init(
-        name: String,
-        destinationPath: AbsolutePath,
-        packageType: PackageType
+    public convenience init(fileSystem: FileSystem,
```

Comment:
is this constructor for testing? if so, it should be moved as an extension to the test module